### ISSUE TITLE
[[TypedArrayConstructorName]] no longer needed

### ISSUE THREAD

**Issue by allenwb** [tc39/ecma262#271]
The [[TypedArrayConstructorName]] internal slot of *TypedArray* constructors is no longer needed now that the secondary prototype walk has been eliminated by the changes made in the 2015-12-20 draft.

[[TypedArrayConstructorName]] was formally needed because each of the individual *TypedArray* constructors `super` called to %TypedArray% and that shared constructor algorithm needed a way to identify the element type of the instance being constructed.

However, with the recent changes, each *TypedArray* constructor is a distinct function.  Hence, the desired element type can be literally coded within each constructor.

Here is sketch of the changes that need to be make:
 *  In [22.2.4.1](https://tc39.github.io/ecma262/#sec-typedarray)
   *  add a new step after step 1: Let *constructorName* be the String value of the Constructor Name value specified in Table 50 for this *TypedArray* constructor.
   *  Change the first argument of the call to **AllocateTypedArray** to be *constructorName*  instead of  *TypedArray*.[[TypedArrayConstructorName]].
 *  Make similar changes in 22.2.4.2, 22.2.4.3, 22.2.4.5
 *  In [22.2.5](https://tc39.github.io/ecma262/#sec-properties-of-the-typedarray-constructors) delete the paragraph that defines the [[TypedArrayConstructorName]] internal slot.

**Comment by ljharb** [comment]
No objections or concerns from me :-) but just in case, I wanted to point to my [which-typed-array](https://github.com/ljharb/which-typed-array) module whose behavior indirectly depends on the `[[TypedArrayName]]` internal slot of instances, which are currently populated via the `[[TypedArrayConstructorName]]` mechanism (and will remain correctly populated in each `AllocateTypedArray` call with @allenwb's suggested changes).
